stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerView.kt

```diff
@@ -48,14 +66,40 @@ public class MessageComposerView : ConstraintLayout {
     private fun init() {
         binding = StreamUiMessageComposerBinding.inflate(streamThemeInflater, this)
         binding.leadingContent.apply {
-            val defaultLeadingContent = MessageComposerDefaultLeadingContent(context)
+            val defaultLeadingContent = MessageComposerDefaultLeadingContent(context).apply {
+                onAttachmentsButtonClick = {
+                    context.getFragmentManager()?.let {
+                        AttachmentSelectionDialogFragment.newInstance(MessageInputViewStyle.createDefault(context))
+                            .apply {
+                                val listener =
+                                    AttachmentSelectionListener { attachments: Set<AttachmentMetaData>, attachmentSource: AttachmentSource ->
+                                        attachments.map { // TODO move to view model
```

Comment:
This will eventually land out of the view, ideally in the view model of attachments picker which doesn't yet exist.

Reply:
Maybe we could reuse this ViewModel from Compose